### DISCUSSION TITLE
Diblis MoY Shader 1.0

### DISCUSSION THREAD

**Discussion by dibli-goost** [bnpr/Malt#531] [Showcase]
**Note: I have not tested any of this on Blender 4.0, as I can't update quite just yet. Things might be broken, I'll update with a 4.0 version when I get the rest of my stuff ready to move.**

This is the Malt rendering setup for my wip Visual Novel, Memories of You. In order for it to work, you need to be on the latest version of Malt's Node Groups branch found here: 

https://github.com/bnpr/Malt/releases/tag/node-groups-latest

### **[Download](https://1drv.ms/u/s!AtM_pc3Q-xDuggn30SXSacTNCegw?e=MfTpOa)** 

=== Included Stuff ===

```
 + Several context-focused mega shaders. For Character Shading, Backgrounds, Foliage, and some VFX stuff.

  + Rendering pipeline. Includes Image-Based Lighting via HDRIs and Screenspace Reflections.

 + Line Art set up for rendering Character and Background Lines.

 + EEVEE voronoi texture, for sparklies.

 + Various GLSL tone-mapping presets, the Uchimura Gran Turismo one is currently exposed in the Render Tree.

 + Dynamic Glitch filtering based on any collection named "Filters". Any object within will do stuff to the content behind it! Sorta wip.
```

Most of the work has already been done by Malt itself. I've just tried to wrangle together a pipeline for my own production. I've tried to lay things out so that it's easier for folks to understand and build off of.

HDRI lighting can be turned off via the Pre Pass Pixel Shader Output, individually per material. They have their own unique normal input so as to not mess with other shading. Glitch filters can similarly be controlled via the Pre Pass. Red = Glitch Overlay, Green = Painterly Chromatic Abberation thing. Screen Space Reflections are controlled via the SSGI Main Pass output. Red = Fac, Green = Diffusion.

You must install the included DiblisMoYPlugin in order for the blend file to load properly. Drop it in a global plugins folder, specify that directory in the BlenderMalt addon preferences, and you should be good to go!

To use in your own projects, link the world from the file. make the world then the render tree local, reload pipeline. Make local further from there to edit the parameters in the render tree.

The source for the IBL and SSGI code: #246 #223

The source for the tone-mapping code: https://github.com/dmnsgn/glsl-tone-map

![image](https://github.com/bnpr/Malt/assets/31767887/29968be7-73e3-4e61-8624-9d8b3a250e81)
![mephel confident interior](https://github.com/bnpr/Malt/assets/31767887/0786c726-4922-4703-8256-a497e85375bb)

**Comment by eballeste** [comment]
the scene looks really cool!

**Reply by negicless** [reply]
NIce work, how did you get the reflection on the floor?

**Reply by dibli-goost** [reply]
The reflections are created using the SSGI code found here: #223 

See the SSGI node tree in the "Memories" main pass to see how the functions are used.

**Reply by negicless** [reply]
Thanks! I will give it a try!
